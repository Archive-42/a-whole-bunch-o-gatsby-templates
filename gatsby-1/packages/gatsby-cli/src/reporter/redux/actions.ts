import { bindActionCreators } from "redux"
import { Dispatch } from "redux"
import { dispatch } from "./"

import * as actions from "./internal-actions"

const boundActions = bindActionCreators<typeof actions, any>(
  actions,
  (dispatch as any) as Dispatch
)

export const createLog = boundActions.createLog as typeof actions.createLog
export const createPendingActivity = boundActions.createPendingActivity as typeof actions.createPendingActivity
export const setStatus = boundActions.setStatus as typeof actions.setStatus
export const startActivity = boundActions.startActivity as typeof actions.startActivity
export const endActivity = boundActions.endActivity as typeof actions.endActivity
export const updateActivity = boundActions.updateActivity as typeof actions.updateActivity
export const setActivityErrored = boundActions.setActivityErrored as typeof actions.setActivityErrored
export const setActivityStatusText = boundActions.setActivityStatusText as typeof actions.setActivityStatusText
export const setActivityTotal = boundActions.setActivityTotal as typeof actions.setActivityTotal
export const activityTick = boundActions.activityTick as typeof actions.activityTick
