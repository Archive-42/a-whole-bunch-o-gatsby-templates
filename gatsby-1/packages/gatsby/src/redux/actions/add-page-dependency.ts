import { store } from "../"

import { createPageDependency as internalCreatePageDependency } from "./internal"

export const createPageDependency = ({
  path,
  nodeId,
  connection,
}: {
  path: string
  nodeId: string
  connection?: string
}): void => {
  const { queries } = store.getState()

  // Check that the dependencies aren't already recorded so we
  // can avoid creating lots of very noisy actions.
  let nodeDependencyExists = false
  let connectionDependencyExists = false
  if (!nodeId) {
    nodeDependencyExists = true
  }
  if (
    nodeId &&
    queries.byNode.has(nodeId) &&
    queries.byNode.get(nodeId)!.has(path)
  ) {
    nodeDependencyExists = true
  }
  if (!connection) {
    connectionDependencyExists = true
  }
  if (
    connection &&
    queries.byConnection.has(connection) &&
    queries.byConnection.get(connection)!.has(path)
  ) {
    connectionDependencyExists = true
  }
  if (nodeDependencyExists && connectionDependencyExists) {
    return
  }

  // It's new, let's dispatch it
  store.dispatch(internalCreatePageDependency({ path, nodeId, connection }))
}
