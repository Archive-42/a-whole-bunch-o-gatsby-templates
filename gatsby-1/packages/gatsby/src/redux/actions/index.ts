import * as internalActions from "./internal"
import { actions as publicActions } from "./public"
import {
  actions as restrictedActions,
  availableActionsByAPI,
} from "./restricted"

export const actions = {
  ...internalActions,
  ...publicActions,
  ...restrictedActions,
}

export const restrictedActionsAvailableInAPI = availableActionsByAPI

export { internalActions, publicActions, restrictedActions }
