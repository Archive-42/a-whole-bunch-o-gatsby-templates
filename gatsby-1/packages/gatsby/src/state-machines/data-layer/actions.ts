import { assign, DoneInvokeEvent, ActionFunctionMap } from "xstate"
import { createGraphQLRunner } from "../../bootstrap/create-graphql-runner"
import reporter from "gatsby-cli/lib/reporter"
import { IDataLayerContext } from "./types"
import { assertStore } from "../../utils/assert-store"
import { GraphQLRunner } from "../../query/graphql-runner"

const concatUnique = <T>(
  array1: Array<T> = [],
  array2: Array<T> = []
): Array<T> => Array.from(new Set(array1.concat(array2)))

export const assignChangedPages = assign<
  IDataLayerContext,
  DoneInvokeEvent<{
    changedPages: Array<string>
    deletedPages: Array<string>
  }>
>((context, event) => {
  return {
    pagesToBuild: concatUnique(context.pagesToBuild, event.data.changedPages),
    pagesToDelete: concatUnique(context.pagesToDelete, event.data.deletedPages),
  }
})

export const assignGraphQLRunners = assign<IDataLayerContext>(
  ({ store, program }) => {
    assertStore(store)
    return {
      gatsbyNodeGraphQLFunction: createGraphQLRunner(store, reporter),
      graphqlRunner: new GraphQLRunner(store, {
        collectStats: true,
        graphqlTracing: program?.graphqlTracing,
      }),
    }
  }
)

export const dataLayerActions: ActionFunctionMap<IDataLayerContext, any> = {
  assignChangedPages,
  assignGraphQLRunners,
}
