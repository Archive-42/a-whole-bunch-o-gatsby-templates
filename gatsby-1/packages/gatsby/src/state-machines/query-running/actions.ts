import { IQueryRunningContext } from "./types"
import {
  DoneInvokeEvent,
  assign,
  ActionFunctionMap,
  AnyEventObject,
} from "xstate"
import { enqueueFlush } from "../../utils/page-data"

export const flushPageData = (): void => {
  enqueueFlush()
}

export const assignDirtyQueries = assign<
  IQueryRunningContext,
  DoneInvokeEvent<any>
>((_context, { data }) => {
  const { queryIds } = data
  return {
    queryIds,
  }
})

export const markSourceFilesDirty = assign<IQueryRunningContext>({
  filesDirty: true,
})

export const markSourceFilesClean = assign<IQueryRunningContext>({
  filesDirty: false,
})

export const trackRequestedQueryRun = assign<
  IQueryRunningContext,
  AnyEventObject
>({
  pendingQueryRuns: (context, { payload }) => {
    const pendingQueryRuns = context.pendingQueryRuns || new Set<string>()
    if (payload?.pagePath) {
      pendingQueryRuns.add(payload.pagePath)
    }
    return pendingQueryRuns
  },
})

export const clearCurrentlyHandledPendingQueryRuns = assign<
  IQueryRunningContext
>({
  currentlyHandledPendingQueryRuns: undefined,
})

export const queryActions: ActionFunctionMap<IQueryRunningContext, any> = {
  assignDirtyQueries,
  flushPageData,
  markSourceFilesDirty,
  markSourceFilesClean,
  trackRequestedQueryRun,
  clearCurrentlyHandledPendingQueryRuns,
}
