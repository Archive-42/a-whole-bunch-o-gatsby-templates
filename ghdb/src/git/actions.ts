import { Observable, Subject } from 'rxjs';
import { filter, map, switchMap } from 'rxjs/operators';
import { PerformClone } from './clone';
import { DoesGitRepoExists } from './repo-exists';
import { ValidateGit } from './validate';

class GithubDatabase {
  public setup(): Observable<string> {
    return ValidateGit().pipe(
      map((message: string) => message),
      switchMap((message: string) => DoesGitRepoExists()),
      filter((doesIt: boolean) => !doesIt),
      switchMap(() => PerformClone())
    );
  }
}

export const githubDb = new GithubDatabase();
