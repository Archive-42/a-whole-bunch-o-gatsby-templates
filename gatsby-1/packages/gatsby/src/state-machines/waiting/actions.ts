import {
  AssignAction,
  assign,
  ActionFunctionMap,
  sendParent,
  AnyEventObject,
  ActionFunction,
} from "xstate"
import { IWaitingContext } from "./types"
import { AnyAction } from "redux"
import { callRealApi } from "../../utils/call-deferred-api"

export const callApi: ActionFunction<IWaitingContext, AnyEventObject> = (
  { store },
  event
) => callRealApi(event.payload, store)

/**
 * Event handler used when we're not ready to process node mutations.
 * Instead we add it to a batch to process when we're next idle
 */
export const addNodeMutation: AssignAction<IWaitingContext, AnyAction> = assign(
  {
    nodeMutationBatch: ({ nodeMutationBatch = [] }, { payload }) => {
      // It's not pretty, but it's much quicker than concat
      nodeMutationBatch.push(payload)
      return nodeMutationBatch
    },
  }
)

export const extractQueries = sendParent<IWaitingContext, AnyEventObject>(
  `EXTRACT_QUERIES_NOW`
)

export const waitingActions: ActionFunctionMap<IWaitingContext, AnyAction> = {
  addNodeMutation,
  extractQueries,
  callApi,
}
