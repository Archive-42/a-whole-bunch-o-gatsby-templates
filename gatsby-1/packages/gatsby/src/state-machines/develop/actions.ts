import {
  assign,
  AnyEventObject,
  ActionFunction,
  spawn,
  ActionFunctionMap,
  DoneEventObject,
} from "xstate"
import { IBuildContext } from "../../services"
import { actions } from "../../redux/actions"
import { listenForMutations } from "../../services/listen-for-mutations"
import { DataLayerResult } from "../data-layer"
import { saveState } from "../../redux/save-state"
import reporter from "gatsby-cli/lib/reporter"
import { store } from "../../redux"
import { ProgramStatus } from "../../redux/types"
import { createWebpackWatcher } from "../../services/listen-to-webpack"
import { callRealApi } from "../../utils/call-deferred-api"
/**
 * Handler for when we're inside handlers that should be able to mutate nodes
 * Instead of queueing, we call it right away
 */
export const callApi: ActionFunction<IBuildContext, AnyEventObject> = (
  { store },
  event
) => callRealApi(event.payload, store)

/**
 * Event handler used in all states where we're not ready to process node
 * mutations. Instead we add it to a batch to process when we're next idle
 */
export const addNodeMutation = assign<IBuildContext, AnyEventObject>({
  nodeMutationBatch: ({ nodeMutationBatch = [] }, { payload }) => {
    // It's not pretty, but it's much quicker than concat
    nodeMutationBatch.push(payload)
    return nodeMutationBatch
  },
})

export const assignStoreAndWorkerPool = assign<IBuildContext, DoneEventObject>(
  (_context, event) => {
    const { store, workerPool } = event.data
    return {
      store,
      workerPool,
    }
  }
)

const setQueryRunningFinished = async (): Promise<void> => {
  store.dispatch(
    actions.setProgramStatus(ProgramStatus.BOOTSTRAP_QUERY_RUNNING_FINISHED)
  )
}

export const markQueryFilesDirty = assign<IBuildContext>({
  queryFilesDirty: true,
})

export const markSourceFilesDirty = assign<IBuildContext>({
  sourceFilesDirty: true,
})

export const markSourceFilesClean = assign<IBuildContext>({
  sourceFilesDirty: false,
})

export const markNodesDirty = assign<IBuildContext>({
  nodesMutatedDuringQueryRun: true,
})

export const markNodesClean = assign<IBuildContext>({
  nodesMutatedDuringQueryRun: false,
})

export const incrementRecompileCount = assign<IBuildContext>({
  nodesMutatedDuringQueryRunRecompileCount: ({
    nodesMutatedDuringQueryRunRecompileCount: count = 0,
  }) => {
    reporter.verbose(
      `Re-running queries because nodes mutated during query run. Count: ${
        count + 1
      }`
    )
    return count + 1
  },
})

export const resetRecompileCount = assign<IBuildContext>({
  nodesMutatedDuringQueryRunRecompileCount: 0,
  nodesMutatedDuringQueryRun: false,
})

export const assignServiceResult = assign<IBuildContext, DoneEventObject>(
  (_context, { data }): DataLayerResult => data
)

/**
 * This spawns the service that listens to the `emitter` for various mutation events
 */
export const spawnMutationListener = assign<IBuildContext>({
  mutationListener: () => spawn(listenForMutations, `listen-for-mutations`),
})

export const assignServers = assign<IBuildContext, AnyEventObject>(
  (_context, { data }) => {
    return {
      ...data,
    }
  }
)

export const spawnWebpackListener = assign<IBuildContext, AnyEventObject>({
  webpackListener: ({ compiler }) => {
    if (!compiler) {
      return undefined
    }
    return spawn(createWebpackWatcher(compiler))
  },
})

export const assignWebhookBody = assign<IBuildContext, AnyEventObject>({
  webhookBody: (_context, { payload }) => payload?.webhookBody,
  webhookSourcePluginName: (_context, { payload }) => payload?.pluginName,
})

export const clearWebhookBody = assign<IBuildContext, AnyEventObject>({
  webhookBody: undefined,
  webhookSourcePluginName: undefined,
})

export const finishParentSpan = ({ parentSpan }: IBuildContext): void =>
  parentSpan?.finish()

export const saveDbState = (): Promise<void> => saveState()

export const logError: ActionFunction<IBuildContext, AnyEventObject> = (
  _context,
  event
) => {
  reporter.error(event.data)
}

export const panic: ActionFunction<IBuildContext, AnyEventObject> = (
  _context,
  event
) => {
  reporter.panic(event.data)
}

export const panicBecauseOfInfiniteLoop: ActionFunction<
  IBuildContext,
  AnyEventObject
> = () => {
  reporter.panic(
    reporter.stripIndent(`
  Panicking because nodes appear to be being changed every time we run queries. This would cause the site to recompile infinitely.
  Check custom resolvers to see if they are unconditionally creating or mutating nodes on every query.
  This may happen if they create nodes with a field that is different every time, such as a timestamp or unique id.`)
  )
}

export const trackRequestedQueryRun = assign<IBuildContext, AnyEventObject>({
  pendingQueryRuns: (context, { payload }) => {
    const pendingQueryRuns = context.pendingQueryRuns || new Set<string>()
    if (payload?.pagePath) {
      pendingQueryRuns.add(payload.pagePath)
    }
    return pendingQueryRuns
  },
})

export const clearPendingQueryRuns = assign<IBuildContext>(() => {
  return {
    pendingQueryRuns: new Set<string>(),
  }
})

export const buildActions: ActionFunctionMap<IBuildContext, AnyEventObject> = {
  callApi,
  markNodesDirty,
  addNodeMutation,
  spawnMutationListener,
  assignStoreAndWorkerPool,
  assignServiceResult,
  assignServers,
  markQueryFilesDirty,
  assignWebhookBody,
  clearWebhookBody,
  finishParentSpan,
  spawnWebpackListener,
  markSourceFilesDirty,
  markSourceFilesClean,
  markNodesClean,
  incrementRecompileCount,
  resetRecompileCount,
  panicBecauseOfInfiniteLoop,
  saveDbState,
  setQueryRunningFinished,
  panic,
  logError,
  trackRequestedQueryRun,
  clearPendingQueryRuns,
}
