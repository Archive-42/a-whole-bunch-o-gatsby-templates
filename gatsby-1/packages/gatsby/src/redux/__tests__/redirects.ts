import { actions } from "../actions"
import { store } from "../index"

jest.mock(`../index`, () => {
  return {
    store: {
      getState: jest.fn(),
    },
    dispath: (): void => {},
    emitter: {
      on: jest.fn(),
    },
  }
})

const protocolArr = [
  [`https`, `https://example.com`],
  [`http`, `http://example.com`],
  [`//`, `//example.com`],
  [`ftp`, `ftp://example.com`],
  [`mailto`, `mailto:example@email.com`],
]

describe(`Add redirects`, () => {
  beforeEach(() => {
    ;(store.getState as jest.Mock).mockReturnValue({
      program: { pathPrefixs: false },
    })
  })

  it(`allows you to add redirects`, () => {
    const action = actions.createRedirect({
      fromPath: `/old/hello-world`,
      toPath: `/new/hello-world`,
    })

    expect(action).toMatchSnapshot()
  })
  it(`create redirects as permanent`, () => {
    const action = actions.createRedirect({
      fromPath: `/old/hello-world`,
      toPath: `/new/hello-world`,
      isPermanent: true,
    })

    expect(action).toMatchSnapshot()
  })

  it(`creates redirects with in-browser redirect option`, () => {
    const action = actions.createRedirect({
      fromPath: `/old/hello-world`,
      toPath: `/new/hello-world`,
      redirectInBrowser: true,
    })

    expect(action).toMatchSnapshot()
  })

  protocolArr.forEach(([protocol, toPath], index) => {
    it(`creates redirects to the URL starts with ${protocol}`, () => {
      const action = actions.createRedirect({
        fromPath: `/old/hello-world-${index}`,
        toPath,
      })

      expect(action).toMatchSnapshot()
    })
  })

  protocolArr.forEach(([protocol, fromPath], index) => {
    it(`creates redirects from the URL starts with ${protocol}`, () => {
      const action = actions.createRedirect({
        fromPath,
        toPath: `/new/hello-world-${index}`,
      })

      expect(action).toMatchSnapshot()
    })
  })
})

describe(`Add redirects with path prefixs`, () => {
  beforeEach(() => {
    ;(store.getState as jest.Mock).mockReturnValue({
      program: {
        prefixPaths: true,
      },
      config: {
        pathPrefix: `/blog`,
      },
    })
  })
  it(`allows you to add redirects`, () => {
    const action = actions.createRedirect({
      fromPath: `/old/hello-world`,
      toPath: `/new/hello-world`,
    })

    expect(action).toMatchSnapshot()
  })
  it(`create redirects as permanent`, () => {
    const action = actions.createRedirect({
      fromPath: `/old/hello-world`,
      toPath: `/new/hello-world`,
      isPermanent: true,
    })

    expect(action).toMatchSnapshot()
  })

  it(`creates redirects with in-browser redirect option`, () => {
    const action = actions.createRedirect({
      fromPath: `/old/hello-world`,
      toPath: `/new/hello-world`,
      redirectInBrowser: true,
    })

    expect(action).toMatchSnapshot()
  })

  protocolArr.forEach(([protocol, toPath], index) => {
    it(`creates redirects to the URL starts with ${protocol}`, () => {
      const action = actions.createRedirect({
        fromPath: `/old/hello-world-${index}`,
        toPath,
      })

      expect(action).toMatchSnapshot()
    })
  })

  protocolArr.forEach(([protocol, fromPath], index) => {
    it(`creates redirects from the URL starts with ${protocol}`, () => {
      const action = actions.createRedirect({
        fromPath,
        toPath: `/new/hello-world-${index}`,
      })

      expect(action).toMatchSnapshot()
    })
  })
})
