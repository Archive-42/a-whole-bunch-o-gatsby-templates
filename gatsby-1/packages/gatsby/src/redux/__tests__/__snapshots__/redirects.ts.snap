// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add redirects allows you to add redirects 1`] = `
Object {
  "payload": Object {
    "fromPath": "/old/hello-world",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/new/hello-world",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects create redirects as permanent 1`] = `
Object {
  "payload": Object {
    "fromPath": "/old/hello-world",
    "ignoreCase": true,
    "isPermanent": true,
    "redirectInBrowser": false,
    "toPath": "/new/hello-world",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects from the URL starts with // 1`] = `
Object {
  "payload": Object {
    "fromPath": "//example.com",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/new/hello-world-2",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects from the URL starts with ftp 1`] = `
Object {
  "payload": Object {
    "fromPath": "ftp://example.com",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/new/hello-world-3",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects from the URL starts with http 1`] = `
Object {
  "payload": Object {
    "fromPath": "http://example.com",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/new/hello-world-1",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects from the URL starts with https 1`] = `
Object {
  "payload": Object {
    "fromPath": "https://example.com",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/new/hello-world-0",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects from the URL starts with mailto 1`] = `
Object {
  "payload": Object {
    "fromPath": "mailto:example@email.com",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/new/hello-world-4",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects to the URL starts with // 1`] = `
Object {
  "payload": Object {
    "fromPath": "/old/hello-world-2",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "//example.com",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects to the URL starts with ftp 1`] = `
Object {
  "payload": Object {
    "fromPath": "/old/hello-world-3",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "ftp://example.com",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects to the URL starts with http 1`] = `
Object {
  "payload": Object {
    "fromPath": "/old/hello-world-1",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "http://example.com",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects to the URL starts with https 1`] = `
Object {
  "payload": Object {
    "fromPath": "/old/hello-world-0",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "https://example.com",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects to the URL starts with mailto 1`] = `
Object {
  "payload": Object {
    "fromPath": "/old/hello-world-4",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "mailto:example@email.com",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects creates redirects with in-browser redirect option 1`] = `
Object {
  "payload": Object {
    "fromPath": "/old/hello-world",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": true,
    "toPath": "/new/hello-world",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs allows you to add redirects 1`] = `
Object {
  "payload": Object {
    "fromPath": "/blog/old/hello-world",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/blog/new/hello-world",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs create redirects as permanent 1`] = `
Object {
  "payload": Object {
    "fromPath": "/blog/old/hello-world",
    "ignoreCase": true,
    "isPermanent": true,
    "redirectInBrowser": false,
    "toPath": "/blog/new/hello-world",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects from the URL starts with // 1`] = `
Object {
  "payload": Object {
    "fromPath": "//example.com",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/blog/new/hello-world-2",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects from the URL starts with ftp 1`] = `
Object {
  "payload": Object {
    "fromPath": "ftp://example.com",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/blog/new/hello-world-3",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects from the URL starts with http 1`] = `
Object {
  "payload": Object {
    "fromPath": "http://example.com",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/blog/new/hello-world-1",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects from the URL starts with https 1`] = `
Object {
  "payload": Object {
    "fromPath": "https://example.com",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/blog/new/hello-world-0",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects from the URL starts with mailto 1`] = `
Object {
  "payload": Object {
    "fromPath": "mailto:example@email.com",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "/blog/new/hello-world-4",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects to the URL starts with // 1`] = `
Object {
  "payload": Object {
    "fromPath": "/blog/old/hello-world-2",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "//example.com",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects to the URL starts with ftp 1`] = `
Object {
  "payload": Object {
    "fromPath": "/blog/old/hello-world-3",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "ftp://example.com",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects to the URL starts with http 1`] = `
Object {
  "payload": Object {
    "fromPath": "/blog/old/hello-world-1",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "http://example.com",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects to the URL starts with https 1`] = `
Object {
  "payload": Object {
    "fromPath": "/blog/old/hello-world-0",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "https://example.com",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects to the URL starts with mailto 1`] = `
Object {
  "payload": Object {
    "fromPath": "/blog/old/hello-world-4",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": false,
    "toPath": "mailto:example@email.com",
  },
  "type": "CREATE_REDIRECT",
}
`;

exports[`Add redirects with path prefixs creates redirects with in-browser redirect option 1`] = `
Object {
  "payload": Object {
    "fromPath": "/blog/old/hello-world",
    "ignoreCase": true,
    "isPermanent": false,
    "redirectInBrowser": true,
    "toPath": "/blog/new/hello-world",
  },
  "type": "CREATE_REDIRECT",
}
`;
